source/frontend/src/components/maps/leaflet/Control/LayersControl/LayersMenu.tsx

```diff
@@ -266,7 +266,12 @@ const LayersMenu: React.FC<React.PropsWithChildren<unknown>> = () => {
         //add any layers defined in the configuration.
         const layer = confLayers?.find(cl => cl.key === parent.key);
 
-        const allNodes = [...(parent.nodes ?? []), ...(layer?.nodes ?? [])];
+        const layerNodes = [...(layer?.nodes ?? [])];
+        const parentNodes =
+          parent?.nodes?.filter(node => !layerNodes.find(layerNode => layerNode.id === node.ids)) ??
```

Comment:
Please change `node.ids` to `node.id` (notice the extra "s" at the end). The correct property is "id".
Also it would be great if unit tests could be added to cover this component as there's no coverage currently.

Reply:
Will do!

Reply:
Hi @buddy326 - are unit tests coming for this changes? or will they be added in a future PR?

Reply:
Hi, yes I can work on those today. I've been so occupied by other tickets that this slipped around the radar. Thanks

Reply:
@buddy326 just a reminder about the requested unit tests on this one. Remember, we like to try and prioritize closing all open PRs before opening new ones.

Reply:
I've been having some trouble setting these up so switching back and forth between tickets I can make better progress on but I will try to get them added in as soon as possible here.

Reply:
Pushing what I do have so far and then I'll work on them some more after preparing my other stories for the demo.